src/DiagramGenerator/Diagram/Board.php

```diff
@@ -127,8 +130,11 @@ public function drawBoard()
 
         // Add board texture
         if ($this->getBoardTexture()) {
-            $background = new \Imagick($this->getBackgroundTexture());
-            $textureSize = $this->getCellSize() * 2;
+            try {
+                $background = $this->getBackgroundTexture();
+            } catch (\ImagickException $exception) {
+                throw new CachedFileInvalidException();
```

Comment:
Can we preserve the original exception message? Just discarding the original exception means we lose a lot of debug info.